### PR TITLE
modernize genkernel build script

### PR DESCRIPTION
**Pull request by timkenhan**
ref: https://github.com/gentoo/genkernel/pull/56

### CODE REVIEW

File: sys-kernel/genkernel/genkernel-9999.ebuild [hunk 1]
```diff
@@ -128,11 +129,9 @@ RDEPEND="${PYTHON_DEPS}
 	dev-build/libtool
 	virtual/pkgconfig
 	elibc_glibc? ( sys-libs/glibc[static-libs(+)] )
-	firmware? ( sys-kernel/linux-firmware )"
+	firmware? ( sys-kernel/linux-firmware )
+"
 
-if [[ ${PV} == 9999* ]]; then
-	DEPEND="${DEPEND} app-text/asciidoc"
-fi
```
Comment (robbat2) [inline sys-kernel/genkernel/genkernel-9999.ebuild:135]: no, this is still required in DEPEND

File: sys-kernel/genkernel/genkernel-9999.ebuild [hunk 2]
```diff
@@ -164,72 +163,20 @@ src_prepare() {
...
-		-e "s:VERSION_ZLIB:${VERSION_ZLIB}:"\
-		-e "s:VERSION_ZSTD:${VERSION_ZSTD}:"\
-		"${S}"/defaults/software.sh \
-		|| die "Could not adjust versions"
+	export VERSION_BCACHE_TOOLS VERSION_BOOST VERSION_BTRFS_PROGS VERSION_BUSYBOX VERSION_COREUTILS
+	export VERSION_CRYPTSETUP VERSION_DMRAID VERSION_DROPBEAR VERSION_EUDEV VERSION_EXPAT VERSION_E2FSPROGS
+	export VERSION_FUSE VERSION_GPG VERSION_HWIDS VERSION_ISCSI VERSION_JSON_C VERSION_KMOD VERSION_LIBAIO
+	export VERSION_LIBGCRYPT VERSION_LIBGPGERROR VERSION_LIBXCRYPT VERSION_LVM VERSION_LZO VERSION_MDADM
+	export VERSION_POPT VERSION_STRACE VERSION_THIN_PROVISIONING_TOOLS VERSION_UNIONFS_FUSE VERSION_USERSPACE_RCU
+	export VERSION_UTIL_LINUX VERSION_XFSPROGS VERSION_XZ VERSION_ZLIB VERSION_ZSTD VERSION_KEYUTILS
```
Comment (robbat2) [inline sys-kernel/genkernel/genkernel-9999.ebuild:171]: Replace the entire mess
```
    # Export all the versions that may be used by genkernel build.
    for v in $(set |awk -F= '/^VERSION_/{print $1}') ; do
        export ${v}
    done
```

File: sys-kernel/genkernel/genkernel-9999.ebuild [hunk 3]
```diff
@@ -164,72 +163,20 @@ src_prepare() {
...
-		emake
-	fi
+	emake PREFIX=/usr
 }
 
 src_install() {
-	insinto /etc
-	doins "${S}"/genkernel.conf
-
-	doman genkernel.8
```
Comment (robbat2) [inline sys-kernel/genkernel/genkernel-9999.ebuild:217]: subtle bug introduced here. If /usr/share/man8 does not exist, the codebase creates /usr/share/man/man8.bz2 with that content.

File: sys-kernel/genkernel/genkernel-9999.ebuild [hunk 4]
```diff
@@ -164,72 +163,20 @@ src_prepare() {
...
-	dodoc AUTHORS ChangeLog README TODO
```
Comment (robbat2) [inline sys-kernel/genkernel/genkernel-9999.ebuild:218]: The Makefile does not install these, please keep this line for now

File: sys-kernel/genkernel/genkernel-9999.ebuild [hunk 5]
```diff
@@ -164,72 +163,20 @@ src_prepare() {
...
-	dobin genkernel
-	rm -f genkernel genkernel.8 AUTHORS ChangeLog README TODO genkernel.conf
-
-	if use ibm ; then
-		cp "${S}"/arch/ppc64/kernel-2.6{-pSeries,} || die
-	else
-		cp "${S}"/arch/ppc64/kernel-2.6{.g5,} || die
-	fi
```
Comment (robbat2) [inline sys-kernel/genkernel/genkernel-9999.ebuild:226]: @gentoo/ppc  Are these files needed at all anymore?

File: sys-kernel/genkernel/genkernel-9999.ebuild (hunk 3 above)
Comment (robbat2) [reply sys-kernel/genkernel/genkernel-9999.ebuild:217]: I fixed it in the main repo

**Comment by robbat2** [comment]
@timkenhan  your signed-off-by is also required in the commit.

File: sys-kernel/genkernel/genkernel-9999.ebuild (hunk 1 above)
Comment (timkenhan) [reply sys-kernel/genkernel/genkernel-9999.ebuild:135]: I've moved it up to the main DEPEND definition [here](https://github.com/gentoo/gentoo/pull/35089/files/1127ffdeea9c6b2120bddbbc1b6e428e0290e94f#diff-9cb3123499b5ef6595e1a2fd0b48c1b758f7789f1f5e46340db76b18baaa4eccR120).

By the way, shouldn't we have this as BDEPEND instead, as it isn't required for runtime?

**Comment by timkenhan** [comment]
> @timkenhan your signed-off-by is also required in the commit.

@robbat2 got it, I'll rebase and redo the commit

File: sys-kernel/genkernel/genkernel-9999.ebuild (hunk 2 above)
Comment (timkenhan) [reply sys-kernel/genkernel/genkernel-9999.ebuild:171]: didn't know this is a thing. thanks for the input!

File: sys-kernel/genkernel/genkernel-9999.ebuild (hunk 1 above)
Comment (robbat2) [reply sys-kernel/genkernel/genkernel-9999.ebuild:135]: yes, BDEPEND and/or DEPEND, but certainly not in RDEPEND where you've put it.

Comment (timkenhan) [reply sys-kernel/genkernel/genkernel-9999.ebuild:135]: I see what you mean now! I'll change it accordingly

**Comment by robbat2** [comment]
Thanks @timkenhan 
I fixed the typo in your signed-off by commit;
p.s. you can use `git commit -s` to signoff without having to add the trailer manually.
